### ISSUE TITLE
Lint against case where a variable name could be a flag

### ISSUE THREAD

**Issue by KSXGitHub** [koalaman/shellcheck#2612]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
set -o errexit -o pipefail -o nounset

printf '%s' "Enter file name to delete... "
read -r filename
rm "$filename"
```

#### Here's what shellcheck currently says:

Nothing.

#### Here's what I wanted or expected to see:

Warns that `$filename` could be a flag, and suggests inserting `--` before `$filename`.
